### PR TITLE
Add Token authentication to REST API 

### LINKED ISSUE #27: Basic Authentication Setup
Using django.contrib.auth
Add endpoint for token request, returning django standard tokens
Include token type into URL to allow for future support of other tokens (?)

Depends on #10

### PR DESCRIPTION
**Pull request by em0lar**
This PR adds token authentication to REST API.
Currently the only way to obtain a token (without being authenticated) is to send a POST request to the API endpoint `/api/authentication/token/credentials` with the following body:
```json
{
  "username": "<youruser>",
  "password": "<password>",
  "application": "Web"
}
```
As response you get the token you can use in the `Authentication` header (example `Authentication Token abc`).

**Future**
I plan several more authentication features:
* Token Authentication via Cookie
* OIDC (Implicit Flow + Authorization Code [maybe with PKCE])

fixes #27

### CODE REVIEW

File: src/chaosinventory/authentication/serializers.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from rest_framework import serializers
+
+from .models import Token
+
+
+class ObtainAuthTokenSerializer(serializers.Serializer):
+    application = serializers.CharField(label="Application Name")
+    expiring = serializers.DateTimeField(
+        label="Date/Time when token expires",
+        required=False)
```
Comment (e1mo) [inline src/chaosinventory/authentication/serializers.py]: The expiry date provided via the API can be in the past, I think if an application implements it it is their fault, but we could also throw an exception.

Comment (e1mo) [inline src/chaosinventory/authentication/serializers.py]:
```suggestion
    expiring = serializers.DateTimeField(
        label="Date/Time when token expires",
        required=False,
    )
```

File: src/chaosinventory/authentication/serializers.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    renewable = serializers.BooleanField(
+        label="Renewability of token",
+        default=True)
```
Comment (e1mo) [inline src/chaosinventory/authentication/serializers.py]:
```suggestion
    expiring = serializers.DateTimeField(
        label="Date/Time when token expires",
        required=False)
    renewable = serializers.BooleanField(
        label="Renewability of token",
        default=True,
    )
```

File: src/chaosinventory/authentication/models.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    application = models.CharField(max_length=255)
+    expiring = models.DateTimeField(
+        null=True,
+        blank=True
+    )
```
Comment (e1mo) [inline src/chaosinventory/authentication/models.py:24]: Currently, it is possible to set expiring dates in the past. Whilst this mostly is an issue for the apps providing a wrong date, we could in the future catch this in the validation process.

**Review by e1mo — APPROVED** [review]
Overall this looks quite good, thanks a lot! My only notable "issue" is, that the fields are currently not really documented in the schema.yml. For that, a custom [Schema definition](https://www.django-rest-framework.org/api-guide/schemas/) is required. However, I think this could be postponed and dealt with in an issue to be created after this PR is merged.

File: doc/dev/api/authentication.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+Authentication
+==============
+
+In order to use the API you need to authenticate yourself. In future there
+will be multiple ways to do use, currently the only option is via an token.
+
+Token Authentication
+--------------------
+
```
Comment (margau) [inline doc/dev/api/authentication.rst:12]: We should add a description which type of tokens and which length is returned around here.

File: src/chaosinventory/authentication/models.py (hunk 3 above)
Comment (margau) [reply src/chaosinventory/authentication/models.py:24]: I'd suggest opening an issue for later validation improvement, as it is no blocking or security relevant issue for now?

File: doc/dev/api/authentication.rst (hunk 4 above)
Comment (e1mo) [reply doc/dev/api/authentication.rst:12]: See 582b539

**Review by margau — APPROVED** [review]

**Pull request merged**